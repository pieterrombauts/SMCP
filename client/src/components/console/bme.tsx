import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import bme_1 from "./../../media/Bme_1.png"
import bme_2 from "./../../media/Bme_2.png"

interface AppProps {};

const bme: React.FC<AppProps> = () => {
  return(
         <Tabs defaultActiveKey="Spartan_console" id="uncontrolled-tab-example"  >
        
          <Tab eventKey="BME_1" title="Astronaut Health">
            <img src={bme_1} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="BME_2" title="Astronaut Vitals">
          <img src={bme_2} style={{width:"300px"}}/> 

          </Tab>
        </Tabs>
  );
}

export const Bme = bme;