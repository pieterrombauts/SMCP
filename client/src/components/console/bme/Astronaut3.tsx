import React from 'react';
import { bmeVitalSignsDisplay } from '../customTypes'

interface A3Props {
  values: bmeVitalSignsDisplay;
};

const Astronaut3: React.FC<A3Props> = ( props ) => {
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="" y="55%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="22%" y="59.5%" width={150} height={150} href='/media/Bme_3.gif'></image>
      <image x="2%" y="52%" width={250} height={500} href='/media/Bme_Christina_Koch.jpg'></image>

      <text x="2%" y="60%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Christina Koch</text>
      <text x="2%" y="95%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 3</text>

      <text x="22%" y="59.5%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="34%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="41%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="34%" y="66.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut3.heartRate}</text>
      <text x="41%" y="66.5%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut3.heartRateMax}</text>

      <text x="22%" y="76.5%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="22%" y="79.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">115 / 70</text>

      <text x="22%" y="84.5%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
    
    </g>
  )
};

export default Astronaut3;