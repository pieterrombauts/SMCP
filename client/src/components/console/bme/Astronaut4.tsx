import React from 'react';
import { bmeVitalSignsDisplay } from '../customTypes'

interface A4Props {
  values: bmeVitalSignsDisplay;
};

const Astronaut4: React.FC<A4Props> = ( props ) => {
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="50%" y="55%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="72%" y="59.5%" width={150} height={150} href='/media/Bme_4.gif'></image>
      <image x="52%" y="52%" width={250} height={500} href='/media/Bme_Jessica_Meir.jpg'></image>

      <text x="52%" y="60%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Jessica Meir</text>
      <text x="52%" y="95%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 5</text>
      
      <text x="72%" y="59.5%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="84%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="91%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="84%" y="66.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut4.heartRate}</text>
      <text x="91%" y="66.5%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut4.heartRateMax}</text>

      <text x="72%" y="76.5%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="72%" y="79.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">115 / 70 </text>

      <text x="72%" y="84.5%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <text x="91%" y="84.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut4.temperature}</text>

    </g>
  )
};

export default Astronaut4;