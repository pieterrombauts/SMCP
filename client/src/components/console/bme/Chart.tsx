import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { Line } from 'react-chartjs-2';
import {chartData} from '../customTypes'

interface ChartProps {
  className?: string;
  data: {x: string, y: number};
};

const UChart: React.FC<ChartProps> = (props) => {
  const [chartData, setChartData] = useState<chartData>({
    datasets: [
      {
        data: [
          {x: '2019-11-15T14:55:00', y: 37.2}
        ],
        backgroundColor: 'white',
        borderColor: 'red',
        borderWidth: 2,
        fill: false,
      }
    ]
  })
  useEffect(() => {
    setChartData({
      datasets: [
        {
          data: [...chartData.datasets[0].data, props.data],
          backgroundColor: 'white',
          borderColor: 'red',
          borderWidth: 2,
          fill: false
        }
      ]
    })
  }, [props.data])
  return (
    <div className={props.className}>
      <Line
        data={chartData}
        width={200}
        height={90}
        redraw={false}
        options={{
          animation: {
            duration: 0
          },
          title: {
            display: false
          },
          legend: {
            display: false
          },
          elements: {
            point: {
              radius: 0
            },
            line: {
              tension: 0
            }
          },
          layout: {
            padding: {
              bottom: 5
            }
          },
          scales: {
            xAxes: [{
              gridLines: {
                color: '#378545',
                lineWidth: 1
              },
              type: 'time',
              ticks: {
                min: '2019-11-15T15:00:00',
                max: '2019-11-15T16:45:00'
              },
              time: {
                unit: 'minute',
                stepSize: 30
              }
            }],
            yAxes: [{
              gridLines: {
                color: '#378545',
                lineWidth: 1
              },
              ticks: {
                min: 35.8,
                max: 37.8,
                stepSize: 0.2
              }
            }]
          },
          bezierCurve: false,
          mainAspectRation: false,
          responsive: true
        }}
      />
    </div>
  )
};

export const Chart = styled(UChart)`
  background-color: black;
  padding: 5px;
`;