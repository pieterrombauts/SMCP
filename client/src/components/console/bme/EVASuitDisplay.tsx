import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import EVA1 from './EVA1';
import EVA2 from './EVA2';
import { generateRandVal } from '../../../dataGenerator';
import { decreaseVal } from '../../../dataDecrease'
import { bmeEVASuitDisplay } from '../customTypes'


interface EVASDProps {
    className?: string;
};

const defaultValues = {
    eva1: {
        helmetPressure: 0,
        o2: 0,
        water: 0,
        batteryCharge: 0
    },

    eva2: {
        helmetPressure: 0,
        o2: 0,
        water: 0,
        batteryCharge: 0
    }
}

const UEVASuitDisplay: React.FC<EVASDProps> = (props) => {
    var [values, setValues] = useState(defaultValues)
    const intervalRef = useRef(0);

    useEffect(() => {
        if (intervalRef.current !== 0) {
            clearInterval(intervalRef.current);
        }

        
        console.log((values.eva1.o2));


        const id = setInterval(() => {
            console.log(values.eva1.o2)
            var newValues =     {
                eva1: {
                    helmetPressure: generateRandVal(8, 10).toFixed(2),
                    o2: generateRandVal(60, 63).toFixed(2),
                    water: generateRandVal(74, 76).toFixed(2),
                    batteryCharge: generateRandVal(47, 45).toFixed(2)
                },

                eva2: {
                    helmetPressure: generateRandVal(8, 10).toFixed(2),
                    o2: generateRandVal(60, 63).toFixed(2),
                    water: generateRandVal(74, 76).toFixed(2),
                    batteryCharge: generateRandVal(47, 45).toFixed(2)
                }

            }
            
            setValues(newValues);
            //console.log((newValues.eva1.o2));
            
            //decreaseVal(newValues.eva1.o2);
            // console.log(("New Values: " + newValues.eva1.o2));
            // decreaseVal(values.eva1.o2);
            // console.log(("Values: " + values.eva1.o2));


        }, 5000);
        intervalRef.current = id;
    }, [])
    return (

        <div className={props.className}>
            <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1400 1000'>
                <rect x="1%" y="5%" rx="5" ry="5" width="1300" height="945" style={{ fill: "#333333", stroke: "#393939", strokeWidth: 5, opacity: 0.5 }} />

                <text x="37%" y="18%" fill="#EBB249" fontSize="30" textDecoration="underline"> Helmet Pressure (psi)</text>
                <line x1="47%" y1="20%" x2="47%" y2="30%" stroke="#a39d9d" strokeWidth="4" />

                <text x="41%" y="34%" fill="#EBB249" fontSize="30" textDecoration="underline"> O2 Level (%)</text>
                <line x1="47%" y1="36%" x2="47%" y2="46%" stroke="#a39d9d" strokeWidth="4" />


                <text x="40%" y="50%" fill="#EBB249" fontSize="30" text-decoration="underline"> Water Level (%)</text>
                <line x1="47%" y1="52%" x2="47%" y2="62%" stroke="#a39d9d" strokeWidth="4" />

                <text x="37%" y="66%" fill="#EBB249" fontSize="30" textDecoration="underline"> Battery Chrage Level (%)</text>
                <line x1="47%" y1="68%" x2="47%" y2="75%" stroke="#a39d9d" strokeWidth="4" />

                <text x="43.5%" y="78%" fill="#EBB249" fontSize="30" textDecoration="underline"> Comms</text>
                <line x1="47%" y1="80%" x2="47%" y2="89%" stroke="#a39d9d" strokeWidth="4" />

                <EVA1 values={values} />
                <EVA2 values={values} />


            </svg>
        </div>
    );
}

export const EVASuitDisplay = styled(UEVASuitDisplay)`
  width: 100%;
  height: 100%;
  position: absolute;

`;