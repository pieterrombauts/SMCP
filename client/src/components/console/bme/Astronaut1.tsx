import React, {useState, useEffect} from 'react';
import { bmeVitalSignsDisplay, bmeVitalSignsTemps } from '../customTypes'
import {Chart} from './Chart';
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';

interface A1Props {
  values: bmeVitalSignsDisplay;
  temps: bmeVitalSignsTemps;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time
})

const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & A1Props

const Astronaut1: React.FC<Props> = ( props ) => {
  const [newData, setNewData] = useState<{x: string, y: number}>({x:"2019-11-15T15:00:00", y:37})
  useEffect(() => {
    if (props.time !== "") setNewData({x: props.time, y: props.temps.astronaut1.temperature})
  }, [props.temps])
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect x="0%" y="3%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="22%" y="7.5%" width={150} height={150} href='/media/Bme_1.gif'></image>
      <image x="2%" y="0%" width={250} height={500} href='/media/Bme_Luca_Parmitano.jpg'></image>
      
      <text x="2%" y="8%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Luca Parmitano</text>
      <text x="2%" y="43%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Commander</text>

      <text x="22%" y="8%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="34%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="41%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="34%" y="15%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.heartRate}</text>
      <text x="41%" y="15%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.heartRateMax}</text>

      <text x="22%" y="25%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="22%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.bloodPressureSystolic.toFixed(0)} / </text>
      <text x="26.5%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.bloodPressureDiastolic.toFixed(0)}</text>

      <text x="22%" y="33%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <foreignObject x="22%" y="34%" height="125px" width="300px" fill="black"><Chart data={newData}/></foreignObject>
    </g>
  )
};

export default connector(Astronaut1);