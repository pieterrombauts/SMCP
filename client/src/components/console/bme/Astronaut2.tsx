import React, {useState, useEffect} from 'react';
import { bmeVitalSignsDisplay, bmeVitalSignsTemps} from '../customTypes'
import {Chart} from './Chart';
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';

interface A2Props {
  values: bmeVitalSignsDisplay;
  temps: bmeVitalSignsTemps;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time
})

const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & A2Props

export const Astronaut2: React.FC<Props> = ( props ) => {
  const [newData, setNewData] = useState<{x: string, y: number}>({x:"2019-11-15T15:00:00", y:37})
  useEffect(() => {
    if (props.time !== "") setNewData({x: props.time, y: props.temps.astronaut2.temperature})
  }, [props.temps])
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="50%" y="3%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="72%" y="7.5%" width={150} height={150} href='/media/Bme_2.gif'></image>
      <image x="52%" y="0%" width={250} height={500} href='/media/Bme_Andrew_R_Morgan.jpg'></image>
      
      <text x="52%" y="8%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Andrew Morgan</text>
      <text x="52%" y="43%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 2</text>

      <text x="72%" y="8%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="84%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="91%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="84%" y="15%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.heartRate}</text>
      <text x="91%" y="15%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.heartRateMax}</text>

      <text x="72%" y="25%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="72%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.bloodPressureSystolic.toFixed(0)} /</text>
      <text x="76.5%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.bloodPressureDiastolic.toFixed(0)}</text>

      <text x="72%" y="33%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <foreignObject x="72%" y="34%" height="125px" width="300px" fill="black"><Chart data={newData}/></foreignObject>
    </g>
  )
};

export default connector(Astronaut2);