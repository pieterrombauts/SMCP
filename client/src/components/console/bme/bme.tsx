import React, { useState } from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav';
import Tab from 'react-bootstrap/Tab';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { VitalSignsDisplay } from './VitalSignsDisplay'
import { EVASuitDisplay } from './EVASuitDisplay'
import socket from '../../Socket'

interface AppProps {
  className?: string;
};

const UBme: React.FC<AppProps> = (props) => {
  const [ firstOpensVS, setFirstOpensVS ] = useState<boolean>(true);
  function updateFirstConsoleVSOpen(event: React.MouseEvent<any>) {
    if (firstOpensVS) {
      socket.emit('FIRST_CONSOLE_OPEN', 'bme-vs')
      setFirstOpensVS(false);
    }
  }
  return (
    <Tab.Container id="bme-tabs" defaultActiveKey="EVASuit">
      <Row className={props.className}>
        <Col id="bme-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="EVASuit">
              <EVASuitDisplay />
            </Tab.Pane>
            <Tab.Pane onClick={updateFirstConsoleVSOpen} eventKey="vitalSigns">
              <VitalSignsDisplay />
            </Tab.Pane>
          </Tab.Content>
        </Col>
        <Col id="bme-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="EVASuit">EVA Suit</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleVSOpen} eventKey="vitalSigns">Vital Signs</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>

  );
}

export const Bme = styled(UBme)`
  width: 100%;
  height: 100%;
  position: absolute;

  #bme-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #bme-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #bme-buttons .nav-item a {
    padding: 20px 30px;
  }
`;