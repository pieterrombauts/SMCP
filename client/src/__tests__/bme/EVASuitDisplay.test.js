import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { UEVASuitDisplay } from '../../components/console/bme/EVASuitDisplay'
import { shallow, mount, render} from 'enzyme';

describe("EVA suit display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UEVASuitDisplay />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})