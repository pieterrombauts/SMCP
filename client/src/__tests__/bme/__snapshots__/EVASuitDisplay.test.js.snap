// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVA suit display renders correctly 1`] = `
<div>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1400 1000"
  >
    <rect
      height="945"
      rx="5"
      ry="5"
      style={
        Object {
          "fill": "#333333",
          "opacity": 0.5,
          "stroke": "#393939",
          "strokeWidth": 5,
        }
      }
      width="1300"
      x="1%"
      y="5%"
    />
    <text
      fill="#EBB249"
      fontSize="30"
      textDecoration="underline"
      x="37%"
      y="18%"
    >
       Helmet Pressure (psi)
    </text>
    <line
      stroke="#a39d9d"
      strokeWidth="4"
      x1="47%"
      x2="47%"
      y1="20%"
      y2="30%"
    />
    <text
      fill="#EBB249"
      fontSize="30"
      textDecoration="underline"
      x="41%"
      y="34%"
    >
       O2 Level (%)
    </text>
    <line
      stroke="#a39d9d"
      strokeWidth="4"
      x1="47%"
      x2="47%"
      y1="36%"
      y2="46%"
    />
    <text
      fill="#EBB249"
      fontSize="30"
      textDecoration="underline"
      x="40%"
      y="50%"
    >
       Water Level (%)
    </text>
    <line
      stroke="#a39d9d"
      strokeWidth="4"
      x1="47%"
      x2="47%"
      y1="52%"
      y2="62%"
    />
    <text
      fill="#EBB249"
      fontSize="30"
      textDecoration="underline"
      x="37%"
      y="66%"
    >
       Battery Charge Level (%)
    </text>
    <line
      stroke="#a39d9d"
      strokeWidth="4"
      x1="47%"
      x2="47%"
      y1="68%"
      y2="75%"
    />
    <text
      fill="#EBB249"
      fontSize="30"
      textDecoration="underline"
      x="43.5%"
      y="78%"
    >
       Comms
    </text>
    <line
      stroke="#a39d9d"
      strokeWidth="4"
      x1="47%"
      x2="47%"
      y1="80%"
      y2="89%"
    />
    <EVA1
      values={
        Object {
          "eva1": Object {
            "batteryCharge": 67,
            "helmetPressure": 0,
            "o2": 65,
            "water": 80,
          },
          "eva2": Object {
            "batteryCharge": 68,
            "helmetPressure": 0,
            "o2": 65,
            "water": 80,
          },
        }
      }
    />
    <EVA2
      values={
        Object {
          "eva1": Object {
            "batteryCharge": 67,
            "helmetPressure": 0,
            "o2": 65,
            "water": 80,
          },
          "eva2": Object {
            "batteryCharge": 68,
            "helmetPressure": 0,
            "o2": 65,
            "water": 80,
          },
        }
      }
    />
  </svg>
</div>
`;
